src/Shared/UnitTests/ObjectModelHelpers.cs

```diff
@@ -1936,5 +1939,126 @@ public void Dispose()
                 _buildManager.Dispose();
             }
         }
+
+        internal class LoggingFileSystem : IMSBuildFileSystem
+        {
+            private readonly IFileSystem _wrappingFileSystem;
+            private int _fileSystemCalls;
```

Comment:
Out of curiosity, why are we only curious (for now) about how many file system calls there have been?

Should there be a test for verifying this increments properly?

Reply:
This class is a mock class for the actual tests, no used in production. It would be a first in the repo to have tests for test helpers, so I'd rather not :)

Reply:
If our issue count dips bellow 500 its time to rewrite the engine to make it saner.